hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -171,7 +172,7 @@ public static SparkConf registerClasses(SparkConf conf) {
       JavaRDD<HoodieRecord<T>> taggedRecords = index.tagLocation(dedupedRecords, jsc, table);
       metrics.updateIndexMetrics(LOOKUP_STR, metrics.getDurationInMs(indexTimer == null ? 0L : indexTimer.stop()));
       indexTimer = null;
-      return upsertRecordsInternal(taggedRecords, commitTime, table, true);
+      return upsertRecordsInternal(taggedRecords, commitTime, table, true, Type.UPSERT);
```

Comment:
There is an enum OperationType in HoodieWriteClient. It is more fine-grained in that it is able to distinguish between PREPPED and non-PREDDED version of operations. Can we use that enum instead of Type. You can move it to a separate enum class in the package org.apache.hudi.common.model and name it as WriteOperationType